test/framework/src/main/java/org/elasticsearch/common/logging/JsonLogsIntegTestCase.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Tests that extend this class verify that all json layout fields appear in the first few log lines after startup
+ * Fields available straight away : type, timestamp, level, component, message, nodeName, clusterName
+ * nodeId and clusterId are available later once the clusterState was received.
+ * NodeName, ClusterName, NodeId, ClusterId should be consistent across all log lines
+ * Note that this won't pass for clusters that don't have
+ * the node name defined in elasticsearch.yml <strong>and</strong> start with
+ * DEBUG or TRACE level logging. Those nodes log a few lines before they
+ * resolve the node name.
+ */
+public abstract class JsonLogsIntegTestCase extends ESRestTestCase {
+    /**
+     * Number of lines in the log file to check for the node name. We don't
+     * just check the entire log file because it could be quite long and
+     * exceptions don't include the node name.
+     */
+    private static final int LINES_TO_CHECK = 10;
```

Comment:
Now that this class isn't explicitly about node names I'd change the name of this constant.

Reply:
the name is good, but the comment needs updating